### PR TITLE
Add an env variable as a rollout flag for CheckAccess

### PR DESCRIPTION
**Pull request by nwnt**
rp-production.json was regenerated with the changes from
- resources_rp.go
- templates_rp.go
- rpVMSS.sh

### Which issue this PR addresses:

Fixes ARO-2922

### What this PR does / why we need it:

CheckAccess migration will only happen gradually starting with INT and Canary regions.  To orchestrate such gradual release, a new environment variable is introduced to signify if the region will use CheckAccess or not.

### Test plan for issue:
Tested this in INT.

### CODE REVIEW

File: pkg/deploy/generator/templates_rp.go
Comment (bennerv) [inline pkg/deploy/generator/templates_rp.go:88]: nit: no extra space.

**Review by bennerv — CHANGES_REQUESTED** [review]
first pass looks great- let's use LiveConfig though so we can leverage it when it's implemented

File: pkg/deploy/generator/templates_rp.go
Comment (bennerv) [inline pkg/deploy/generator/templates_rp.go:88]: nit: won't need this space after we use liveconfig

**Review by SudoBrendan — APPROVED** [review]
Overall LGTM! A few nit-picks you can take or dismiss :)

**Review by bennerv — APPROVED** [review]
lgtm - if you want to make the changes suggested by Brendan, let me know after and I'll approve + merge again.

File: pkg/api/validate/dynamic/dynamic.go
Comment (nwnt) [reply pkg/api/validate/dynamic/dynamic.go]: Done

File: pkg/api/validate/dynamic/dynamic_test.go
Comment (nwnt) [reply pkg/api/validate/dynamic/dynamic_test.go]: Done!  I extracted the variables into corresponding *var* blocks.

**Review by bennerv — DISMISSED** [review]
lgtm

**Review by bennerv — APPROVED** [review]

File: pkg/util/liveconfig/checkaccess.go
Comment (petrkotas) [inline pkg/util/liveconfig/checkaccess.go:1]: Please document what checkAccess is. There is no explanation of what this option does and what are the implications of using it.

**Review by petrkotas — DISMISSED** [review]
@nwnt Thank you for the PR, it is a clean work. Nicely done. I have one simple ask. Please add a explanation of what CheckAccessV2 is, what does it do and what are the implications of using it. If we do not write it down now it will end up as a tribal knowledge that will be known only to few.
Which will eventually create bottle neck for fixing issues in the code. Thank you! 

PS: I looked at attached Jira tickets, there is no explanation there either.

File: pkg/api/validate/dynamic/dynamic.go
Comment (s-amann) [inline pkg/api/validate/dynamic/dynamic.go:472]: is there value in keeping this log here: `c.dv.log.Infof("%s has no access to %s", *c.oid, result.ActionId)`

**Review by s-amann — COMMENTED** [review]
one question, otherwise lgtm.

One other concern, but I'm not sure if this would happen with 'real' data. When looping through the results, if we don't find a matching action within the results, we will stop looping over actions as `found` will never be set to true.

File: pkg/util/liveconfig/checkaccess.go
Comment (nwnt) [reply pkg/util/liveconfig/checkaccess.go:1]: Hi @petrkotas, definitely.  I have some info in the ADO wiki [here](https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/AzureRedHatOpenShift.wiki/409184/CheckAccess-API-migration).  Let me know if this is ok or if there are any points missing.

I should also be in a habit of including more info on the jira child tickets.  There are some background info on the parent one in ARO-1800.  Let me know if you have any suggestion on how I can improve it.

File: pkg/api/validate/dynamic/dynamic.go
Comment (nwnt) [reply pkg/api/validate/dynamic/dynamic.go:472]: I'm springing a lot of logging statements here because the API hasn't been GAed yet and there are chances that it might give wrong results.  I plan to remove most of the loggings once we have completely switched over to the new API.  For now, I think the more data to observe how the API behaves, the better.

For the looping question, if we don't find a matching action, then it means the particular action/permission hasn't been granted to the service principal.  In that case, it should stop and go into another retry call.

**Comment by bennerv** [comment]
@petrkotas  if the doc links Nont provided are good enough (note they're in Jira as well), please feel free to give this a thumbs up and merge.  I'm good with it as is.

**Review by jhoreman — APPROVED** [review]
Thanks Nont, lgtm!

**Review by s-amann — APPROVED** [review]

**Comment by bennerv** [comment]
@petrkotas If you look at the epic it has links to MSFT documentation.  A tl;dr: We currently use an older API to check permissions against identities.  There is a new API that MSFT is currently releasing called CheckAccess, which will have better "freshness" (cache) times and it is handled in a different manner.  It's currently not fully GA in all of public cloud, but undergoing rollout and one of our canary regions is swapped over to the new API to confirm it is working as expected.  

Going to merge.

**Pull request merged**
